### PR TITLE
Added option to aws_acm_facts module to search by ARN instead of just name

### PR DESCRIPTION
**Pull request by stefanhorning**
##### SUMMARY
As the ARN is the only unique identifier and is also the only search param the AWS API takes it makes highly sense to allow it as a param for this module too.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
aws_acm_facts module

##### ADDITIONAL INFORMATION
Just passes through the `certificate_arn` to the already existing get and describe methods.

### CODE REVIEW

**Comment by stefanhorning** [comment]
ready_for_review

**Comment by stefanhorning** [comment]
ready_for_review

**Comment by stefanhorning** [comment]
Any reason not to merge this?

**Comment by stefanhorning** [comment]
@willthames any chance to get this reviewed/merged?

**Comment by willthames** [comment]
LGTM, just would be good to have someone independently test that it works.

Or have a test suite (feature requests really should have a test suite)

**Comment by stefanhorning** [comment]
Cool, I will add the a test suite then.

**Comment by stefanhorning** [comment]
Although there is hardly anything I can test, as there seems to be no ACM module to create certs, hence I can't search and expect for specific ones. Also it would probably need a CI update (which usually never happens) to run them as part of he build.

**Comment by stefanhorning** [comment]
Anything else I can do to move this forward?

File: test/integration/targets/aws_acm_facts/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+- block:
+    - name: Get all ACM certificate facts
+      aws_acm_facts:
+        <<: *aws_connection_info
+      register: result
+
+    - assert:
+        that:
+          - result.changed
```
Comment (willthames) [inline test/integration/targets/aws_acm_facts/tasks/main.yml]: I don't expect a facts module to change things.

**Review by willthames — COMMENTED** [review]
Does this module really return changed=True?

**Comment by willthames** [comment]
It would be good to at least have setup instructions for the test suite if the test suite can't itself create certs - there's no easy way for me to independently run this to check that it works

File: test/integration/targets/aws_acm_facts/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      aws_access_key: "{{ aws_access_key }}"
+      aws_secret_key: "{{ aws_secret_key }}"
+      security_token: "{{ security_token }}"
+      region: "{{ aws_region }}"
+  no_log: yes
+
+# Preconditions to run this tests:
+# 1. Create a TLS certificate on AWS ACM (Amazon Certificate Manager), Cosole link: https://console.aws.amazon.com/acm/home
+# 2. Set the domain name in ../defaults/main.yml to the one of your certificate
+# 3. Set the ARN in ../defaults/main.yml to the one of your certificate
```
Comment (stefanhorning) [inline test/integration/targets/aws_acm_facts/tasks/main.yml]: @willthames Will this be sufficient?

File: test/integration/targets/aws_acm_facts/tasks/main.yml (hunk 1 above)
Comment (stefanhorning) [reply test/integration/targets/aws_acm_facts/tasks/main.yml]: Fixed. You are right of course.

**Comment by stefanhorning** [comment]
`aws_acm_facts` has been renamed to `aws_acm_info` in the meantime. Updated this PR accordingly.

**Comment by stefanhorning** [comment]
I think this PR is no longer needed since https://github.com/ansible/ansible/pull/60552 has been merged adding ARN filtering (among many other things)

**Pull request closed without merging**